reactor-core/src/main/java/reactor/core/publisher/CallSiteSupplierFactory.java

```diff
@@ -0,0 +1,169 @@
+/*
+ * Copyright (c) 2023 VMware Inc. or its affiliates, All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package reactor.core.publisher;
+
+import java.util.function.Supplier;
+import java.util.stream.Stream;
+
+import sun.misc.JavaLangAccess;
+import sun.misc.SharedSecrets;
+
+import static reactor.core.publisher.Traces.full;
+import static reactor.core.publisher.Traces.isUserCode;
+import static reactor.core.publisher.Traces.shouldSanitize;
+
+class CallSiteSupplierFactory implements Supplier<Supplier<String>> {
+
+	static final Supplier<Supplier<String>> supplier;
+
+	static {
+		String[] strategyClasses = {
+				CallSiteSupplierFactory.class.getName() + "$SharedSecretsCallSiteSupplierFactory",
+				CallSiteSupplierFactory.class.getName() + "$ExceptionCallSiteSupplierFactory",
+		};
+		// find one available call-site supplier w.r.t. the jdk version to provide
+		// linkage-compatibility between jdk 8 and 9+
```

Comment:
Why do we need this logic? As I understand, for JDK9+, the jar would simply use the `StackWalker`. Here for JDK8, we either use `SharedSecrets` or a simple `Exception`, right? We could document in which situation `SharedSecrets` can be unavailable then, but mentioning 9+ here is confusing.

Reply:
outdated comment. I will update this one. however, since both SharedSecrets and default strategy is available in JDK  8 and `SharedSecrets` is much more performant compares to throwing an exception (although not always available) the only way to go is doing reflection